Sources_v3/StreamChatUI/ChatChannel/ChatMessageMetadataView.swift

```diff
@@ -26,10 +40,67 @@ open class ChatMessageMetadataView<ExtraData: UIExtraDataTypes>: View {
     }
 
     override open func setUpLayout() {
-        embed(timestampLabel)
+        stack.addArrangedSubview(currentUserVisabilityIndicator)
+        stack.addArrangedSubview(timestampLabel)
+        embed(stack)
     }
 
     override open func updateContent() {
         timestampLabel.text = message?.createdAt.getFormattedDate(format: "hh:mm a")
+        currentUserVisabilityIndicator.isVisible = message?.onlyVisibleForCurrentUser ?? false
+    }
+}
+
+open class ChatMessageOnlyVisibleForCurrentUserIndicator: View {
+    // MARK: - Subviews
+
+    public private(set) lazy var stack: UIStackView = {
+        let stack = UIStackView()
```

Comment:
Use `.init(arrangedSubviews:)`

Reply:
Adding subviews/arranged subviews is a layout operations so `setUpLayout` is the right place for then IMO

Reply:
I see your point.
IMHO, it's redundant pedantry.

But up to you.